tap_rest_api_msdk/client.py

```diff
@@ -16,7 +16,13 @@ class RestApiStream(RESTStream):
     # Intialise self.http_auth used by prepare_request
     http_auth = None
     # Cache the authenticator using a Smart Singleton pattern
-    _authenticator = None
+    try:
+        self.assigned_authenticator
```

Comment:
The linters are choking on using `self` outside of methods. There are a lot of other linting errors I'm getting as well. You can run `poetry run tox -e py` to check them and resolve them.

Reply:
Still working on this one.

Reply:
I believe the linters are all happy now. I refactored the code slightly which has the benefit of better practices and making the linters happy.

I also bumped the SDK to the latest version while I was there. I note there is a deprecated warning related to fs which is used by the SDK - at this stage it hasn't been addressed by Meltano. I get a clean run from the linters however.

Reply:
All the issues should be resolved now. Ready for a re-review. Thank you 👍